pkg/runner/logger.go

```diff
@@ -64,30 +72,51 @@ func WithJobLogger(ctx context.Context, jobName string, secrets map[string]strin
 	return common.WithLogger(ctx, rtn)
 }
 
-type stepLogFormatter struct {
-	color           int
-	secrets         map[string]string
-	insecureSecrets bool
-	masks           *[]string
+func isJSONFormatter(ctx context.Context) bool {
+	if ctxLogger := common.Logger(ctx); ctxLogger != nil {
+		if logger, ok := ctxLogger.(*logrus.Logger); ok {
+			_, ok := logger.Formatter.(*logrus.JSONFormatter)
+			return ok
+		}
+	}
+
+	_, ok := logrus.StandardLogger().Formatter.(*logrus.JSONFormatter)
+	return ok
```

Comment:
This reads like a hack...

Reply:
Any idea how to check what logger the context logger has?

Reply:
I would add a parameter to WithJobLogger, then pass the whole runcontext.Config object or just the jsonLogger field.
We can also keep it like this, I have approved your changes already.